From 92a59ec22b6b4c097c6aee6898071d66b8ce381c Mon Sep 17 00:00:00 2001
From: luofeng <luofeng13@huawei.com>
Date: Mon, 11 Mar 2024 12:15:11 +0800
Subject: [PATCH] support clang build

---
 Make.rules | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Make.rules b/Make.rules
index f1ddc25..fa7c662 100644
--- a/Make.rules
+++ b/Make.rules
@@ -36,10 +36,10 @@ endif
 
 
 $(call set, CROSS, )
-$(call set, CC, $$(CROSS)gcc)
+$(call set, CC, $$(CROSS)$(CC))
 $(call set, AS, $$(CROSS)as)
 $(call set, LD, $$(CROSS)ld)
-$(call set, DEPEND, $$(CROSS)gcc -MM -MG)
+$(call set, DEPEND, $$(CROSS)$(CC) -MM -MG)
 $(call set, AR, $$(CROSS)ar)
 $(call set, RANLIB, $$(CROSS)ranlib)
 $(call set, INSTALL, cp)
-- 
2.19.1

