diff --git a/libsupp/clif.c b/libsupp/clif.c
index 2c21326..6e92a26 100644
--- a/libsupp/clif.c
+++ b/libsupp/clif.c
@@ -398,7 +398,7 @@ int CLIF_parse_cmdline (int argc, char *argv[],
 	unsigned int dirty_flags = 0;
 	int dirty_plus = 0;
 	int exclusive_cnt = 0;
-	int posix = getenv ("POSIXLY_CORRECT") != NULL ||
+	int posix = secure_getenv ("POSIXLY_CORRECT") != NULL ||
 					    (parse_flags & CLIF_POSIX);
 
 	curr.argc = argc;
diff --git a/traceroute/as_lookups.c b/traceroute/as_lookups.c
index a9962be..980d91c 100644
--- a/traceroute/as_lookups.c
+++ b/traceroute/as_lookups.c
@@ -39,10 +39,10 @@ const char *get_as_path (const char *query) {
 	    struct addrinfo *res;
 	    int ret;
 
-	    server = getenv ("RA_SERVER");
+	    server = secure_getenv ("RA_SERVER");
 	    if (!server)  server = DEF_RADB_SERVER;
 
-	    service = getenv ("RA_SERVICE");
+	    service = secure_getenv ("RA_SERVICE");
 	    if (!service)  service = DEF_RADB_SERVICE;
 
 
